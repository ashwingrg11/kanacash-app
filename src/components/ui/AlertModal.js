import React from 'react';
import {
  View,
  StyleSheet,
  Modal,
  TouchableWithoutFeedback,
  TouchableOpacity,
} from 'react-native';
import {SemiBoldText, RegularText} from '~/components/ui/Text';
import theme from '~/components/theme/Style';

export default function AlertModal({
  visible,
  onRequestClose,
  onPressOk,
  title,
  message,
  closeButton,
}) {
  return (
    <Modal
      statusBarTranslucent={true}
      animationType="fade"
      transparent={true}
      onRequestClose={onRequestClose}
      visible={visible}>
      <TouchableWithoutFeedback style={styles.mainView}>
        <View style={styles.mainView}>
          <View style={styles.messageWrapper}>
            <View style={styles.alertMessage}>
              {title && <SemiBoldText text={title} invert />}
              <RegularText text={message} style={styles.alertMessageText} />
            </View>

            {!closeButton ? (
              <View style={styles.rowGroup}>
                <TouchableOpacity
                  onPress={onPressOk}
                  style={styles.singleBtnView}>
                  <SemiBoldText text="Yes" invert />
                </TouchableOpacity>

                <TouchableOpacity
                  onPress={onRequestClose}
                  style={styles.cancelButton}>
                  <SemiBoldText text="No" />
                </TouchableOpacity>
              </View>
            ) : (
              <TouchableOpacity
                onPress={onPressOk}
                style={styles.singleBtnView}>
                <SemiBoldText text={closeButton} invert />
              </TouchableOpacity>
            )}
          </View>
        </View>
      </TouchableWithoutFeedback>
    </Modal>
  );
}

const styles = StyleSheet.create({
  mainView: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: 'rgba(0,0,0,0.44)',
  },
  alertMessage: {
    // paddingLeft: 20,
    // paddingRight: 20,
    marginBottom: 20,
    alignItems: 'center',
    justifyContent: 'center',
  },
  messageWrapper: {
    width: '80%',
    backgroundColor: '#1F1F1F',

    padding: 20,
    borderRadius: 5,
  },
  alertMessageText: {
    marginTop: 10,
    fontSize: 16,
    color: '#fff',
    textAlign: 'center',
  },
  cancelButton: {
    justifyContent: 'center',
    alignItems: 'center',
    paddingVertical: 2,
    paddingHorizontal: 40,
    backgroundColor: '#fff',
    borderRadius: 50,
  },
  singleBtnView: {
    justifyContent: 'center',
    alignItems: 'center',
    paddingVertical: 2,
    paddingHorizontal: 40,
    backgroundColor: theme.red,
    borderColor: theme.red,
    borderRadius: 50,
  },
  rowGroup: {
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
});
