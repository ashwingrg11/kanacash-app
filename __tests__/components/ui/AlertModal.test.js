import 'react-native';
import React from 'react';

import AlertModal from '~/components/ui/AlertModal';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

describe('<AlertModal/>', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <AlertModal visible={true}
        title="title"
        message="message"
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
