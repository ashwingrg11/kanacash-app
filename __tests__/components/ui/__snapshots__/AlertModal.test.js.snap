// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AlertModal/> renders correctly 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  statusBarTranslucent={true}
  transparent={true}
  visible={true}
>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(0,0,0,0.44)",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#1F1F1F",
          "borderRadius": 5,
          "padding": 20,
          "width": "80%",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "marginBottom": 20,
          }
        }
      >
        <Text
          invert={true}
          style={
            Array [
              Object {
                "fontFamily": "OpenSans-SemiBold",
                "fontSize": 20,
                "lineHeight": 25,
              },
              Object {
                "color": "#fff",
              },
              undefined,
            ]
          }
          text="title"
        >
          title
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#5C5C5C",
                "fontFamily": "OpenSans-Regular",
                "fontSize": 16,
                "lineHeight": 20,
              },
              undefined,
              Object {
                "color": "#fff",
                "fontSize": 16,
                "marginTop": 10,
                "textAlign": "center",
              },
            ]
          }
          text="message"
        >
          message
        </Text>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <View
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#B31F31",
              "borderColor": "#B31F31",
              "borderRadius": 50,
              "justifyContent": "center",
              "opacity": 1,
              "paddingHorizontal": 40,
              "paddingVertical": 2,
            }
          }
        >
          <Text
            invert={true}
            style={
              Array [
                Object {
                  "fontFamily": "OpenSans-SemiBold",
                  "fontSize": 20,
                  "lineHeight": 25,
                },
                Object {
                  "color": "#fff",
                },
                undefined,
              ]
            }
            text="Yes"
          >
            Yes
          </Text>
        </View>
        <View
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "borderRadius": 50,
              "justifyContent": "center",
              "opacity": 1,
              "paddingHorizontal": 40,
              "paddingVertical": 2,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "OpenSans-SemiBold",
                  "fontSize": 20,
                  "lineHeight": 25,
                },
                undefined,
                undefined,
              ]
            }
            text="No"
          >
            No
          </Text>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;
